### PR TITLE
New features add

### PR DESCRIPTION
**Pull request by ViachaslauKabak**
* Add Sonarqube configuration
* Add Bitbucket configuration
* Add shared libraries configuration
* Add credentials types
* Add default plugins
* Add custom files copy
* Add initial jobs seed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,10 +206,111 @@ jenkins2_credentials:
...
+    type: 'password'
+    id: 'pipeline_libraries'
+    description: 'username and password for pipeline libraries'
+    username: 'pipelineLibrariesUser'
+    password: 'pipelineLibrariesPassword'
+  sshconnection:
+    type: 'key'
+    keySource: 0
+    key: >
+      -----BEGIN RSA PRIVATE KEY-----
```
Comment (vutkin) [inline README.md]: Should we store ssh private key in public?

File: defaults/main.yml [hunk 2]
```diff
@@ -99,3 +106,151 @@ jenkins2_credentials:
...
+    type: 'password'
+    id: 'pipeline_libraries'
+    description: 'username and password for pipeline libraries'
+    username: 'pipelineLibrariesUser'
+    password: 'pipelineLibrariesPassword'
+  sshconnection:
+    type: 'key'
+    keySource: 0
+    key: >
+      -----BEGIN RSA PRIVATE KEY-----
```
Comment (vutkin) [inline defaults/main.yml]: Should we store private key here?

File: meta/main.yml [hunk 3]
```diff
@@ -2,11 +2,11 @@
 galaxy_info:
   role_name: "jenkins"
   author: "Lean Delivery team <team@lean-delivery.com>"
-  description: "Lean Delivery Jenkins installation and configuration"
+  description: "Jenkins installation and configuration"
   company: "Epam Systems"
   license: "Apache"
   min_ansible_version: "2.5"
-  issue_tracker_url: "https://github.com/lean-delivery/ansible-role-jenkins/issues"
+  issue_tracker_url: "https://github.com/lean-delivery/ansible-role-jenkins2/issues"
```
Comment (vutkin) [inline meta/main.yml]: link `https://github.com/lean-delivery/ansible-role-jenkins/issues2` returns 404
Is it ok?

File: defaults/main.yml (hunk 2 above)
Comment (ViachaslauKabak) [reply defaults/main.yml]: it is just an example

File: README.md (hunk 1 above)
Comment (ViachaslauKabak) [reply README.md]: it is just as an example

File: meta/main.yml (hunk 3 above)
Comment (ViachaslauKabak) [reply meta/main.yml]: fixed

**Comment by abdennour** [comment]
The besT PR so far

**Comment by pavelpikta** [comment]
> The best PR so far. If you could give example of setting a list of jobs to be initialized with the project. I am not talking about seed job, I am talking about jobs itself.

Hi @abdennour . See more in this [template](https://github.com/lean-delivery/ansible-role-jenkins/blob/master/templates/dsl_hybris_create_job.groovy.j2)

**Comment by abdennour** [comment]
awesome! my environment is: Bitbucket Server + Blue Ocean

**Pull request merged**